docs/user-guide/rug/tests.md

```diff
@@ -201,48 +358,220 @@ Given("a Java source file", (p: Project) => {
-The typical approach in a `#!gherkin When` step is to use the
-`#!typescript ProjectScenarioWorld` to look up the Rug you want to run
-and run it.  Here's an example of using a generator.
+```typescript
+import * as stub from "@atomist/cortex/stub/Types";
+import { When, EventHandlerScenarioWorld } from "@atomist/rug/test/handler/Core";
+
+When("an initial commit is received", (w: EventHandlerScenarioWorld) => {
+    const event = new stub.Push()
+        .addCommits(new stub.Commit().withMessage("initial commit by Atomist"))
+        .withRepo(new stub.Repo().withName(repoName).withOwner(orgName)
+            .withOrg(new stub.Org().withOwner(orgName)
+                .withChatTeam(new stub.ChatTeam().withId("NONESUCH").withName("chatter").withProvider("Slack"))));
+    w.sendEvent(event);
+});
+```
+
+### Modifying projects
+
+The "Quick Overview" example above showed how the `#!typescript
+ProjectScenarioWorld` is used to look up and execute an editor.  The
+approach is similar for generators.
 
 ```typescript
 import { Project } from "@atomist/rug/model/Project";
-import { Given, ProjectScenarioWorld } from "@atomist/rug/test/project/Core";
+import { When, ProjectScenarioWorld } from "@atomist/rug/test/project/Core";
 
-When("a generator is run", (p, world) => {
-    let psworld = world as ProjectScenarioWorld;
-    let generator = psworld.generator("TypeScriptGenerator");
-    psworld.generateWith(generator, "new-test-project", {});
+When("project generated", (p: Project, w: ProjectScenarioWorld) => {
+    const generator = w.generator("SomeGenerator");
+    w.generateWith(generator, "new-project", { param1: "value1" });
 });
 ```
 
-The `#!typescript generateWith()` method on `#!typescript
-ProjectScenarioWorld` takes three arguments.  The first is the
-generator to run, which is the valued returned from the `#!typescript
-ProjectScenarioWorld.generator()` method, which finds generators by
-name.  The second argument is the name of the project to be created.
-The third argument is a JSON representation of the generator
-parameters.  In the above example, the generator requires no
-parameters so an empty object is provided.
+The generator is looked up by name using the `generator` method on
+`#!typescript ProjectScenarioWorld`, providing the same name used as
+the first argument to the `#!typescript @Generator` decorator.  The
+generator is executed using the `generateWith` method.  The
+`generateWith` method takes three arguments: the generator, the name
+of the project to be generated, and a JavaScript object whose keys are
+the names of the generator parameters and whose values are the values
+of those parameters.
+
+### Failure
+
+If you want to simulate failure or discover in your `#!gherkin Where`
+step that your world is in an invalid state, the `#!typescript
+ScenarioWorld` interface provides a method to abort the scenario.  If
+you are testing handlers, using the `#!typescript abort` method would
+look like this:
```

Comment:
Could we decribe what is the outcome of this? Would the CLI stop running that test only or all of them?